draco/core/io.py

```diff
@@ -204,6 +204,101 @@ def next(self):
         return map_stack
 
 
+class LoadFITSCatalog(task.SingleTask):
+    """Load an SDSS-style FITS source catalog.
+
+    Catalogs are given as one, or a list of `File Groups` (see
+    :mod:`draco.core.io`). Catalogs within the same group are combined together
+    before being passed on.
+
+    Attributes
+    ----------
+    catalogs : list or dict
+        A dictionary specifying a file group, or a list of them.
+    z_range : list, optional
+        Select only sources with a redshift within the given range.
+    freq_range : list, optional
+        Select only sources with a 21cm line freq within the given range. Overrides
+        `z_range`.
+    """
+
+    catalogs = config.Property(proptype=_list_of_filegroups)
+    z_range = config.list_type(type_=float, length=2, default=None)
+    freq_range = config.list_type(type_=float, length=2, default=None)
+
+    def process(self):
+        """Load the groups of catalogs from disk, concatenate them and pass them on.
+
+        Returns
+        -------
+        catalog : :class:`containers.SpectroscopicCatalog`
+        """
+
+        from astropy.io import fits
+        from . import containers
+
+        # Exit this task if we have eaten all the file groups
+        if len(self.catalogs) == 0:
+            raise pipeline.PipelineStopIteration
+
+        group = self.catalogs.pop(0)
+
+        # Set the redshift selection
+        if self.freq_range:
+            zl = 1420.0 / self.freq_range[1] - 1
+            zh = 1420.0 / self.freq_range[0] - 1
+            self.z_range = (zl, zh)
```

Comment:
May want to replace 1420 with `cora.units.nu21` here for consistency with the beamform and stacking tasks.  This will shift range by ~1 chime frequency bin.

Reply:
Fair point.